### DISCUSSION TITLE
Type Hints or Library Stubs

### DISCUSSION THREAD

**Discussion by gao462** [pyg-team/pytorch_geometric#3429] [Ideas]
Is there a plan to have `torch_geometric` being fully type hinted or provide library stubs?

I want to pass the static typing check (e.g., mypy, pytype) without adding `# type: ignore` to `torch_geometric`.

**Comment by rusty1s** [comment]
I would say this is probably done to 70%. If you are interested, we can finish the remaining part together.

**Reply by rusty1s** [reply]
Started to work on this here: https://github.com/pyg-team/pytorch_geometric/pull/3461

**Reply by mihaic** [reply]
As far as I understand, you need to ship a `py.typed` file with PyG to enable type checking:
https://mypy.readthedocs.io/en/stable/installed_packages.html#creating-pep-561-compatible-packages

Also, the current Mypy test is failing (but appears to be passing because `continue-on-error: true`), e.g.:
https://github.com/pyg-team/pytorch_geometric/actions/runs/5874955029/job/15930495343#step:5:13

**Reply by akihironitta** [reply]
@mihaic Thank you for pointing it out. Will be addressed via #8254.

**Reply by jonasjuerss** [reply]
Sorry for the late follow-up @akihironitta, but was adding ``py.typed`` actually ever addressed? I found that I had to manually:

```toml
[[tool.mypy.overrides]]
module = ["torch_geometric.*"]
follow_untyped_imports = true
```

in my `pyproject.toml`

**Reply by mdbenito** [reply]
Indeed, the file is not there. AFAICT it was removed from #8254 in commit d52ba53705056045b5736df22868734b476bd416 by @rusty1s, maybe by mistake?

**Reply by antspy** [reply]
Hi, 

is there any updates on this? Type stubs do not seem to work atm
